### PR TITLE
migrations: use go1.16 embed directive instead of go-bindata

### PR DESCRIPTION
**Pull request by keegancsmith**
We can remove the bindata generate step, and instead rely on the go compiler to
embed the migrations. Additionally I took the opportunity to remove some
copy-pasta code and just have one migrations package.

### CODE REVIEW

**Comment by keegancsmith** [comment]
@mrnugget test failure is a random batch changes failures. Looks like you need to sort something to make comparison deterministic:

```
batches_test.go:199:   &types.BatchChangesUsageStatistics{
... // 18 identical fields
CurrentMonthContributorsCount: 1,
CurrentMonthUsersCount:        2,
BatchChangesCohorts: []*types.BatchChangesCohort{
- 		&{
- 			Week:                     "2021-01-18",
- 			BatchChangesOpen:         1,
- 			ChangesetsImported:       1,
- 			ChangesetsPublished:      3,
- 			ChangesetsPublishedOpen:  2,
- 			ChangesetsPublishedDraft: 1,
- 		},
&{Week: "2021-03-15", BatchChangesClosed: 1, ChangesetsImported: 1, ChangesetsUnpublished: 2, ...},
+ 		&{
+ 			Week:                     "2021-01-18",
+ 			BatchChangesOpen:         1,
+ 			ChangesetsImported:       1,
+ 			ChangesetsPublished:      3,
+ 			ChangesetsPublishedOpen:  2,
+ 			ChangesetsPublishedDraft: 1,
+ 		},
},
}
```

gonna retry CI

**Comment by keegancsmith** [comment]
ok still failing CI. It seems there have been some commits to main since I created this branch that hopefully fix the flakey test.

**Comment by mrnugget** [comment]
> It seems there have been some commits to main since I created this branch that hopefully fix the flakey test.

Yep, test is fixed on `main`.

**Review by mrnugget — APPROVED** [review]
Nice!

**Review by eseliger — APPROVED** [review]

**Comment by keegancsmith** [comment]
should of checked for other PRs first :D https://github.com/sourcegraph/sourcegraph/pull/19256

File: migrations/frontend/migrations_test.go [hunk 1]
```diff
@@ -1,53 +0,0 @@
...
-	"strconv"
-	"strings"
-	"testing"
-
-	migrations "github.com/sourcegraph/sourcegraph/migrations/frontend"
-)
-
-const FirstMigration = 1528395733
-
-func TestIDConstraints(t *testing.T) {
```
Comment (efritz) [inline migrations/frontend/migrations_test.go:16]: These tests still seem useful though, no?

Comment (eseliger) [reply migrations/frontend/migrations_test.go:16]: whoups you're right

Comment (keegancsmith) [reply migrations/frontend/migrations_test.go:16]: yup which is why I combined all the duplicated tests into one here https://github.com/sourcegraph/sourcegraph/pull/19256/files/45f8cc6adac5629e687b9136b7a7f77d9d2ee340#diff-ebcdb91edd2242775a0eb930982f832616ad008a68ea5fd5f6676ca33cad289bR10

**Pull request merged**